test/install.bats

```diff
@@ -1,7 +1,60 @@
 #!/usr/bin/env bats
 
+setup() {
+  ASDF_HASHICORP="$(dirname "$BATS_TEST_DIRNAME")"
+  # If we put this in $BATS_RUN_TMPDIR, no teardown is required since it's
+  # handled by bats.
+  ASDF_TMPDIR="$(mktemp -p "${BATS_RUN_TMPDIR}" -d "bats-asdf.XXXXXXXXXX")"
```

Comment:
Pass -t to mktemp to create a temporary directory under the /tmp directory.

Reply:
Oh, yay, this is another OS incompatibility between Linux and BSD-derived OSes (e.g., macOS).

GNU `mktemp` deprecates `-t`, instead preferring `-p`:

```
       -p DIR, --tmpdir[=DIR]
              interpret TEMPLATE relative to DIR; if DIR is not  specified,
              use  $TMPDIR  if  set, else /tmp.  With this option, TEMPLATE
              must not be an absolute name; unlike with  -t,  TEMPLATE  may
              contain slashes, but mktemp creates only the final component

       -t     interpret  TEMPLATE as a single file name component, relative
              to a directory: $TMPDIR, if set; else the directory specified
              via -p; else /tmp [deprecated]
```

while BSD `mktemp` only supports `-t`.

Reply:
Hmm, okay let’s just go with -p :p